src/metabase/api/search.clj

```diff
@@ -401,42 +408,51 @@
-                             [:archived-string {:optional true} [:maybe ms/BooleanString]]
-                             [:created-by      {:optional true} [:maybe ms/PositiveInt]]
-                             [:limit           {:optional true} [:maybe ms/Int]]
-                             [:offset          {:optional true} [:maybe ms/Int]]
-                             [:table-db-id     {:optional true} [:maybe ms/PositiveInt]]]]
+           table-db-id
+           verified]}      :- [:map {:closed true}
+                               [:search-string                    [:maybe ms/NonBlankString]]
+                               [:models                           [:maybe [:set SearchableModel]]]
+                               [:archived        {:optional true} [:maybe :boolean]]
+                               [:created-by      {:optional true} [:maybe ms/PositiveInt]]
+                               [:limit           {:optional true} [:maybe ms/Int]]
+                               [:offset          {:optional true} [:maybe ms/Int]]
+                               [:table-db-id     {:optional true} [:maybe ms/PositiveInt]]
+                               [:verified        {:optional true} [:maybe true?]]]]
+  (when (some? verified)
+    (premium-features/assert-has-any-features
+     [:content-verification :official-collections]
+     (deferred-tru "Content Management or Official Collections")))
   (let [models (if (string? models) [models] models)
         ctx    (cond-> {:search-string      search-string
                         :current-user-perms @api/*current-user-permissions-set*
-                        :archived?          (Boolean/parseBoolean archived-string)
+                        :archived?          (boolean archived)
                         :models             models}
                  (some? created-by)  (assoc :created-by created-by)
                  (some? table-db-id) (assoc :table-db-id table-db-id)
                  (some? limit)       (assoc :limit-int limit)
-                 (some? offset)      (assoc :offset-int offset))]
+                 (some? offset)      (assoc :offset-int offset)
+                 (some? verified)    (assoc :verified verified))]
     (assoc ctx :models (search.filter/search-context->applicable-models ctx))))
 
 (api/defendpoint GET "/models"
   "Get the set of models that a search query will return"
-  [q archived-string table-db-id created_by]
-  {archived-string [:maybe ms/BooleanString]
-   table-db-id     [:maybe ms/PositiveInt]
-   created_by      [:maybe ms/PositiveInt]}
-  (query-model-set (search-context {:search-string   q
-                                    :archived-string archived-string
-                                    :table-db-id     table-db-id
-                                    :created-by      created_by
-                                    :models          search.config/all-models})))
+  [q archived table-db-id created_by verified]
```

Comment:
It looks like it's `archived-string` in the docs - I can't seem to find any place where we use the `api/search/models` endpoints instead of just `/api/search` either. But in any case we should be consistent

Reply:
cool. For visibility: I chat with Oisin and looks like we haven't used this param anywhere, and we agree that we'll go with `archived` for both APIs